pages/addUnicorn.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+    <div class="add-unicorn">
+        <formulaire-unicorn :valeur="add" @act="addUnicorn" class="add-unicorn__formulaire-unicorn--padding-top"/>
+    </div>
+</template>
+<script>
+export default{
+    async asyncData( { error, $http }){
+        try{
+            const unicorn = await $http.$get('http://localhost:1337/unicorns/1');
+            return { unicorn }
+        }
+        catch{
+            error({statutCode:404});
+        }
+    },
+
+    computed:{
+        add(){
+            return "add";
+        }
+    },
+    methods: {
+        async addUnicorn(){
+            let object = {};
+                let elm = document.getElementsByTagName("input");
+                for( let i = 0; i<elm.length;i++){
+                    if(elm[i].value !== "" && elm[i].type != 'file') object[elm[i].name] = elm[i].value;
+                }
+                if(elm[4].value !== ""){
```

Comment:
tu peux faire ça plus simplement je pense. Pour cela, tu dois dans le composant formulaire-unicorn, au niveau de l'input type file (dans lequel on peut ajouter une image) ajouter un @change="addImage" et ajouter une méthode addImage dans ce composant qui fait : 

```
addImage (event) {
  const files = event.target.files
      if (files && files.length > 0) {
          this.unicorn.photo = files[0]
      }
 }
```

Tu peux donc supprimer le code des lignes 25 à 36 comprises, et juste faire ta requête  await this.$http.$post("http://localhost:1337/unicorns/", unicorn);